### ISSUE TITLE
Create frontend interface for updating users information.

### ISSUE THREAD

**Issue by wreality** [MESH-Research/CCR#132]
**Problem/Why**
---

Once a user has created an account with a username, email address, and name they cannot make changes to those fields.

**Goal/What**
---

A user should be able to update their username, email address, name, and password while logged into their account.

**Solution/How**
---

Implement front-end pages that allow the user to update the identified information.

### To Do / Acceptance Criteria
-   [x] Edit `pages/Profile.vue` to implement UI
-   [x] Add functionality to frontend for updating user account
-   [x] Write Jest test specs for Profile.vue and AccountLayout.vue
-   [x] Enable Cypress testing for `/account/profile`
-   [x] Change validation on backend to allow updates with duplicate data

#### Bonus
-   [ ] Ensure sending of user updates avoids sending duplicate data requests

### Review Checklist
-   A user can update their:
  - [ ] Username
  - [ ] Password
  - [ ] Email address
  - [ ] Name
-   [ ] Appropriate unit and e2e tests exist to validate the feature

### Accessibility Checklist
-   [ ]  Appropriate aria tags on all form elements.

**Comment by gmeben** [comment]
Currently struggling to solve the problem of updating user data with duplicate data, or data that isn't any different from data that currently exists for a user.

I'm seeing two potential strategies:

1. Update backend to allow duplicate data update requests
    - Relevant documentation: https://lighthouse-php.com/5.2/security/validation.html#validator-classes
2. Update frontend to avoid sending duplicate data update requests

**Comment by gmeben** [comment]
In today's team collaboration time, we decided strategy 3 is best. 

With my colleague's help, I made a breakthrough today getting strategy 1 implemented. 

Now I'm working on getting strategy 2 implemented. There's _many_ ways of doing this. 

Some problems:

-  The objects on the frontend that represent the existing saved state of user data and the state of user data in the page form have values that seem to infinitely expand when inspected in the developer tools.
- Comparing these objects with lodash's `isEqual` behaves unexpectedly.
